distribution/lib/Standard/Database/0.0.0-dev/src/Internal/IR/SQL_Expression.enso

```diff
@@ -47,4 +47,8 @@ type SQL_Expression
          dialect.
        - expressions: a list of expressions which are arguments to the operation
          different operations support different amounts of arguments.
-    Operation (kind : Text) (expressions : Vector SQL_Expression)
+       - metadata: additional metadata tied to the operation. This will be
+         `Nothing` for most operations, but some operations that need to be
+         parametrized by additional settings can use this field to pass that
+         information to the code generator.
+    Operation (kind : Text) (expressions : Vector SQL_Expression) (metadata : Any | Nothing = Nothing)
```

Comment:
Could we have a more restrictive type than `Any | Nothing`? Such as `Row_Number_Metadata | Date_Period_Metadata`? Or do you think there will be too many?

Reply:
We could probably.

I'm not sure if this gives us much benefit - since the 'right' type depends on the particular `kind` anyway, this will not be checking much. We definitely want stronger typing in the new IR, but with this one which is rather dynamic, I'd stick to it I think.

Ideally we should have one parent 'class' or interface for the metadata, but due to lack of inheritance/interfaces we cannot have it. The sum type could work, but I'm afraid it could become long - ideally we should define an alias in the file with metadata and only reference the alias here - but type aliases are not supported yet.